test/test_nn.py

```diff
@@ -1915,7 +1915,7 @@ def _test_gumbel_softmax_st(self, cuda, dtype=torch.float):
 
         exceed_limits = 0
         for draw in range(num_draws):
-            logits_var = torch.tensor(logits, requires_grad=True)
+            logits_var = logits.requires_grad_()
```

Comment:
The code doesn't make a lot of sense now. Can't you just call this line before the loop?